realm/realm-library/src/main/java/io/realm/RealmResults.java

```diff
@@ -138,6 +143,279 @@ public boolean load() {
     }
 
     /**
+     * Sets the value to {@code null} for the given field in all of the objects in the collection.
+     *
+     * @param fieldName name of the field to update.
+     * @throws IllegalArgumentException if field name doesn't exist or isn't nullable.
+     * @throws RealmException if the field is a {@link io.realm.annotations.PrimaryKey} field.
+     */
+    public void setNull(@Nonnull String fieldName) {
+        verifyString(fieldName);
+        realm.checkIfValid();
+        collection.setNull(getColumnIndex(fieldName));
+    }
+
+
+    /**
+     * Sets the {@code boolean} value of the given field in all of the objects in the collection.
+     *
+     * @param fieldName name of the field to update.
+     * @param value new value for the field.
+     * @throws IllegalArgumentException if field name doesn't exist or field isn't boolean.
+     */
+    public void setBoolean(@Nonnull String fieldName, @Nullable Boolean value) {
+        verifyString(fieldName);
+
+        realm.checkIfValid();
+
+        if (checkAndSetNullValue(fieldName, value)) { return; }
+
+        collection.setBoolean(getColumnIndex(fieldName), value);
```

Comment:
I'm not 100% sure, but If we are going to use the `Results` class in Object Store, I'm guessing that is going to take the fieldname as input to it's methods instead of a column index, so `Results.setBoolean(fieldName, value)`. I believe OS already have functionality for mapping between a field name and the index.